### DISCUSSION TITLE
Server search error - e.forEach is not a function

### DISCUSSION THREAD

**Discussion by jadersbr** [sa-si-dev/virtual-select#196] [General]
Hi everyone,

I'm trying to make the search server works, but I have the following error.

`Uncaught (in promise) TypeError: e.forEach is not a function`

Below is my code. Is it a bug or am I doing something wrong?

```
<script>
    VirtualSelect.init({
        ele: '#game_id',
        options: [],
        multiple: true,
        search: true,
        placeholder: "{{__('Select data')}}",
        noOptionsText: "{{__('No results found')}}",
        onServerSearch: getData,
    });

    function getData (searchValue, virtualSelect) {
        let endpoint = @if(isset($endpoint)) @json($endpoint) @else null @endif;
        fetch(endpoint + '?term=' + searchValue).then(function(newOptions) {
            virtualSelect.setServerOptions(newOptions);
        });
    }    
</script>
```

Could you help me?

Regards.

**Comment by sa-si-dev** [comment]
@jadersilvabr it seems you have passed a non-array value like string to `setServerOptions()` method.

I have handled this case in the [release v1.0.35](https://github.com/sa-si-dev/virtual-select/releases/tag/v1.0.35)
